consensus/consensus_leader.go

```diff
@@ -31,13 +31,17 @@ var (
 
 // WaitForNewBlock waits for the next new block to run consensus on
 func (consensus *Consensus) WaitForNewBlock(blockChannel chan *types.Block, stopChan chan struct{}, stoppedChan chan struct{}, startChannel chan struct{}) {
+	first := true
```

Comment:
Add a comment for why adding this boolean?

Reply:
will do.